projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-wpcom-offline-subscription-service.php

```diff
@@ -52,6 +53,57 @@ public function visitor_can_view_content( $valid_plan_ids, $access_level ) {
 		return $this->validate_subscriptions( $valid_plan_ids, $subscriptions );
 	}
 
+	/**
+	 * Check if the subscriber can receive the newsletter
+	 *
+	 * @param int $user_id User id.
+	 * @param int $blog_id Blog id.
+	 * @param int $post_id Post id.
+	 *
+	 * @return bool
+	 * @throws \Exception Throws an exception when used outside of WPCOM.
+	 */
+	public static function subscriber_can_receive_post_by_mail( $user_id, $blog_id, $post_id ) {
+
+		// Site admins can do everything
+		if ( current_user_can( 'edit_post', $post_id ) ) {
+			return true;
+		}
+
+		switch_to_blog( $blog_id );
+
+		$previous_user = wp_get_current_user();
+		wp_set_current_user( $user_id );
+
+		/**
+		 * Filter the subscriptions attached to a specific user on a given site.
+		 *
+		 * @since 9.4.0
+		 *
```

Comment:
I think we should put some of this content in the `visitor_can_view_content` in this class.
I imagine WPCOM_Offline_Subscription_Service being used in email and reader and visitor_can_view_content is probably reader

Reply:
Ok

Reply:
I used `WPCOM_Offline_Subscription_Service::visitor_can_view_content()` here. I had overlooked that method.